packages/mobile/src/transactions/CommentSection.tsx

```diff
@@ -1,17 +1,29 @@
 import HorizontalLine from '@celo/react-components/components/HorizontalLine'
 import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
-import React from 'react'
+import React, { useMemo } from 'react'
 import { useTranslation } from 'react-i18next'
 import { StyleSheet, Text, View } from 'react-native'
+import { decryptComment } from 'src/identity/commentEncryption'
+import useSelector from 'src/redux/useSelector'
+import { dataEncryptionKeySelector } from 'src/web3/selectors'
 
 interface Props {
   comment?: string | null
+  isSend?: boolean
 }
 
-export default function CommentSection({ comment }: Props) {
+export default function CommentSection({ comment, isSend }: Props) {
   const { t } = useTranslation()
-  if (!comment) {
+  const dek = useSelector(dataEncryptionKeySelector)
+  const decryptedComment = useMemo(
+    () =>
+      isSend === undefined
+        ? comment
+        : decryptComment(comment ?? null, dek, isSend)?.comment ?? comment,
+    [comment]
+  )
+  if (!decryptedComment) {
```

Comment:
Not fully understanding why we now need to do this here. Could you explain?

Reply:
Yeah, so before the decryption was being done when navigating to the review screen. I just thought that was a weird place to do the decryption but maybe this isn't the best either 🤔 It's just that since this is where we actually show the comment I figured having the check here made sense. Maybe it would be better to add it as a transformation as soon as we get the comments from the blockchain-api? So we know that the comment is always already decrypted when we need it.

<img width="636" alt="Screen Shot 2022-01-14 at 15 52 00" src="https://user-images.githubusercontent.com/6062888/149569392-cade7d5a-b1f7-4099-9f4b-0b52e6b048c0.png">